### PR TITLE
Add smoke and illumination rounds to D30

### PR DESCRIPTION
**Pull request by joshvelson**
Small change that allows the use of CUP cluster smoke rounds and cluster illum rounds from D30.  Tested fairly extensively, see tac_co_vae_victa_5 (forthcoming, talk to Jack).  Smoke rounds and illum rounds do not have the whistling effect that HE does, but otherwise function correctly.

### CODE REVIEW

File: scripts/fn_d30Strike.sqf [hunk 1]
```diff
@@ -49,3 +52,20 @@ for "_i" from 0 to _barrages - 1 do {
...
+    if (_amount == 0) then {
+        _amount = random 8 + 1;
+    };
+
+    [{
+        params ["_mortar", "_randomPosition", "_ammo", "_amount"];
+        _mortar doArtilleryFire [_randomPosition, _ammo, _amount];
+        _mortar setVehicleAmmo 1;
+    }, [_mortar, _randomPosition, _ammo, _amount], _i * FIREMISSION_DELAY] call CBA_fnc_waitAndExecute;
+};
```
Comment (TyroneMF) [inline scripts/fn_d30Strike.sqf:71]: Why is this in there twice?

Comment (joshvelson) [reply scripts/fn_d30Strike.sqf:71]: It shouldn't be.  Was just using copy-paste on the website editor.

**Review by TyroneMF — APPROVED** [review]

File: scripts/fn_d30Strike.sqf [hunk 2]
```diff
@@ -24,9 +26,10 @@
  * Examples:
  * [D30_1, ["Marker1", "Marker2"], 0, 0, 2] call TAC_Scripts_fnc_d30Strike;
  * [D30_1, tac_MarkerArray, 0, 0, 1] call TAC_Scripts_fnc_d30Strike;
+ *
  */
 
-#define AMMO_TYPES ["CUP_30Rnd_122mmHE_D30_M"]
+#define AMMO_TYPES ["CUP_30Rnd_122mmHE_D30_M","CUP_30Rnd_122mmSMOKE_D30_M","CUP_30Rnd_122mmILLUM_D30_M"]
```
Comment (jonpas) [inline scripts/fn_d30Strike.sqf:32]:
```suggestion
#define AMMO_TYPES ["CUP_30Rnd_122mmHE_D30_M", "CUP_30Rnd_122mmSMOKE_D30_M", "CUP_30Rnd_122mmILLUM_D30_M"]
```

File: scripts/fn_d30Strike.sqf [hunk 3]
```diff
@@ -7,8 +7,10 @@
  * If amount fired is 0 the amount is randomly selected between 1-8
  * Marker array can be set in init.sqf (See Example 2)
  * Marker array is randomly selected but all rounds fire on the same marker.
- * Ammo types are listed as HE (0).
+ * Ammo types are listed as HE (0), Cluster Smoke (1), Illumination (2).
+ * Note that smoke and illumination rounds do not have whistling sound effects.
  * Barrage will have mortars select a new target for every round fired.
+ * Do not use more than 3 cluster smoke rounds or graphics will cut smoke effedts.
```
Comment (jonpas) [inline scripts/fn_d30Strike.sqf:13]:
```suggestion
 * Do not use more than 3 cluster smoke rounds or graphics will cut smoke effects.
```

File: scripts/fn_d30Strike.sqf [hunk 4]
```diff
@@ -24,9 +26,10 @@
  * Examples:
  * [D30_1, ["Marker1", "Marker2"], 0, 0, 2] call TAC_Scripts_fnc_d30Strike;
  * [D30_1, tac_MarkerArray, 0, 0, 1] call TAC_Scripts_fnc_d30Strike;
+ *
```
Comment (jonpas) [inline scripts/fn_d30Strike.sqf:29]:
```suggestion
```

**Pull request merged**